packages/world-vercel/src/streamer.ts

```diff
@@ -1,34 +1,47 @@
 import type { Streamer } from '@workflow/world';
 import { type APIConfig, getHttpConfig, type HttpConfig } from './utils.js';
 
-function getStreamUrl(name: string, httpConfig: HttpConfig) {
+function getStreamUrl(
+  name: string,
+  runId: string | undefined,
+  httpConfig: HttpConfig
+) {
+  if (runId) {
+    return new URL(
+      `${httpConfig.baseUrl}/v1/runs/${runId}/stream/${encodeURIComponent(name)}`
+    );
+  }
   return new URL(`${httpConfig.baseUrl}/v1/stream/${encodeURIComponent(name)}`);
 }
 
 export function createStreamer(config?: APIConfig): Streamer {
   return {
-    async writeToStream(name, chunk) {
+    async writeToStream(
+      name: string,
+      runId: string,
+      chunk: string | Uint8Array
+    ) {
       const httpConfig = await getHttpConfig(config);
-      await fetch(getStreamUrl(name, httpConfig), {
+      await fetch(getStreamUrl(name, runId, httpConfig), {
         method: 'PUT',
         body: chunk,
         headers: httpConfig.headers,
         duplex: 'half',
       });
     },
 
-    async closeStream(name) {
+    async closeStream(name: string) {
       const httpConfig = await getHttpConfig(config);
       httpConfig.headers.set('X-Stream-Done', 'true');
-      await fetch(getStreamUrl(name, httpConfig), {
+      await fetch(getStreamUrl(name, undefined, httpConfig), {
```

Comment:
`undefined` here doesn't seem correct? I think we want `runId` specified here as well.